cpp/src/link_prediction/overlap_impl.cuh

```diff
@@ -24,8 +24,9 @@
 namespace cugraph {
 namespace detail {
 
+template <typename weight_t, bool order_two>
 struct overlap_functor_t {
-  template <typename weight_t>
+  static constexpr bool is_order_two = order_two;
```

Comment:
Do you mean as a runtime variable?

Reply:
It can either be template parameter or a runtime variable... depends on the runtime overhead.

May better be template parameter there as this doesn't affect code size/compile time (order_two is fixed for each similarity type, right?) and this variable is queried inside the innermost loop.

Reply:
And we know order_two value when we call detail::similarity and this parameter is used in detail::similarity. Better directly pass in detail::similarity than indirectly passing through a functor (and the functor does not use this parameter internally...),

Reply:
Changed to pass it as runtime param.